<launch>
    <arg name="nao_ip"              default="192.168.1.113" />
    <arg name="nao_port"            default="9559" />
    <arg name="roscore_ip"          default="192.168.1.112" />
    <arg name="network_interface"   default="wlp3s0" />
    <arg name="namespace"           default="helena" />

        <param name="pepper_ip"             value="$(arg nao_ip)" />
        <param name="pepper_port"           value="$(arg nao_port)" />
        <param name="pepper_name"           value="$(arg namespace)" />

  <!-- Call Robot publisher -->
  <include file="$(find pepper_description)/launch/pepper_upload.launch"/>


  <!-- Call Robot Driver -->
  <node pkg="naoqi_dcm_driver" type="naoqi_dcm_driver" name="naoqi_helena" respawn="false" output="screen" >
    <!-- Load configurations from YAML file to parameter server -->
    <rosparam file="$(find pepper_dcm_bringup)/config/pepper_dcm.yaml" command="load"/>
    <rosparam file="$(find pepper_control)/config/pepper_trajectory_control.yaml" command="load"/>

    <param name="RobotIP"             value="$(arg nao_ip)" />
    <param name="RobotPort"           value="$(arg nao_port)" />
    <param name="DriverBrokerIP"      value="$(arg roscore_ip)" />
    <param name="network_interface"   value="$(arg network_interface)" />
    <param name="Prefix"              value="$(arg namespace)" />
    <param name="motor_groups"        value="Head LArm RArm" /> <!-- either "Body" or separate groups in the order "Head LArm RArm"-->
    <param name="use_dcm"             value="false" />
    <param name="max_stiffness"       value="0.9" />
  </node>

    <node pkg="rur_ros" type="asr_tts_server_2.py" name="asr_tts_helena" respawn="false" output="screen" />

    <node pkg="rur_ros" type="basic_awareness_server_2.py" name="basic_awareness_helena" respawn="false" output="screen"/>
   
    <node pkg="rur_ros" type="behaviours_server_2.py" name="behaviours_helena" respawn="false" output="screen" />
    
    <node pkg="rur_ros" type="pose_server_2.py" name="pose_helena" respawn="false" output="screen" />

    <node pkg="rur_ros" type="audio_server_2.py" name="audio_helena" respawn="false" output="screen" />

    <node pkg="rur_ros" type="sound_localization_server_2.py" name="sound_localization_helena" respawn="false" output="screen" />

</launch>
